pkg/loadbalancer/experimental/healthserver.go

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package experimental
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"strconv"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	"github.com/cilium/statedb"
+
+	lb "github.com/cilium/cilium/pkg/loadbalancer"
+	"github.com/cilium/cilium/pkg/rate"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+// healthServerCell implements the support for the HealthCheckNodePort field.
+// For each service that has HealthCheckNodePort set to a non-zero value it
+// starts an HTTP server on that port and responds with the number of active
+// backends. This allows external load-balancers to avoid directing traffic
+// to services that have no available backends.
```

Comment:
From my understanding of the feature:

```suggestion
// For each service that has HealthCheckNodePort set to a non-zero value it
// starts an HTTP server on that port and responds with the number of active
// backends on the node. This allows external load-balancers to avoid directing traffic
// to nodes that have no available backends.
```